include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/rand/Philox/PhiloxBaseTraits.hpp>
+#include <alpaka/rand/Philox/mulhilo.hpp>
+
+#include <utility>
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) || defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+#    define CUDA_OLD_COMPAT
```

Comment:
That is dangerous, because someone could include `alpaka/rand/Philox/helpers/cuintArray.hpp`. If this is really needed, maybe set it in cmake for the entirety of alpaka.

Reply:
You're right. The cuintArray is still a temporary idea. It would simplify the implementation and possibly help to get a cleaner API, but I don't know if there are any performance or other unforeseen implications.